haystack_experimental/components/tools/openai/function_caller.py

```diff
@@ -0,0 +1,59 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import json
+from typing import List
+
+from haystack import component
+from haystack.dataclasses import ChatMessage
+
+
+@component
+class OpenAIFunctionCaller:
+    """
+    The OpenAIFunctionCaller expects a list of ChatMessages and if there is a tool call with a function name and arguments, it runs the function and returns the
+    result as a ChatMessage from role = 'function'
+    """
+
+    def __init__(self, available_functions):
```

Comment:
Yes we should document the expected format for `available_functions`
```suggestion
    def __init__(self, available_functions: Dict[str, Callable[...]):
```